src/abstractdataframe/reshape.jl

```diff
@@ -394,12 +573,12 @@ Base.reverse(v::RepeatedVector) = RepeatedVector(reverse(parent(v)), v.inner, v.
 Base.similar(v::RepeatedVector, T::Type, dims::Dims) = similar(parent(v), T, dims)
 Base.unique(v::RepeatedVector) = unique(parent(v))
 
-function CategoricalArrays.CategoricalArray(v::RepeatedVector)
-    res = CategoricalArray(parent(v), levels=levels(parent(v)))
-    res.refs = repeat(res.refs, inner = [v.inner], outer = [v.outer])
-    res
-end
-
+# TODO: @nalimilan: is there a generic way to support this?
```

Comment:
@nalimilan - and here I am not sure what would be best. Maybe you can suggest something?

Reply:
I don't see a solution here, except defining an interface in DataAPI. But it's probably not worth it: I suspect very few people use `RepeatedVector`, and even fewer call `CategoricalVector` on it.

One way to make this faster would be to implement the `refarray`/`refvalue`/`refpool` API, which would be useful for other cases too. But this can be done later if somebody cares about it.

Reply:
OK - let us leave it "as is". I expect that `RepeatedVector` is rare enough in practice (and the only negative effect is different order which is not the end of the world I think).